internal/services/sql/sql_server_resource.go

```diff
@@ -315,3 +422,96 @@ func flattenAzureRmSqlServerIdentity(identity *sql.ResourceIdentity) []interface
 
 	return []interface{}{result}
 }
+
+func flattenSqlServerThreatDetectionPolicy(d *pluginsdk.ResourceData, policy sql.ServerSecurityAlertPolicy) []interface{} {
+
+	properties := policy.SecurityAlertPolicyProperties
+
+	securityAlertPolicy := make(map[string]interface{})
+
+	securityAlertPolicy["state"] = string(properties.State)
+	securityAlertPolicy["email_account_admins"] = *properties.EmailAccountAdmins
```

Comment:
there's a crash here fwiw:

```suggestion
	emailAccountAdmins := false
	if properties.EmailAccountAdmins != nil {
		emailAccountAdmins = *properties.EmailAccountAdmins
	}
	securityAlertPolicy["email_account_admins"] = emailAccountAdmins
```